str/associatr/helper/snp_vcf_for_associatr.py

```diff
@@ -0,0 +1,112 @@
+snp_vcf_for_associatr.py --chromosomes=20 --vcf-dir=gs://cpg-bioheart-test/saige-qtl/input_files/genotypes/vds1-0
+
+"""
+
+import gzip
+
+import click
+
+from cpg_utils import to_path
+from cpg_utils.config import output_path
+from cpg_utils.hail_batch import get_batch
+
+
+def reformat_vcf(vcf_file_path, output_file_path):
+    vcf_file = to_path(vcf_file_path)
+    output_file = to_path(output_file_path)
+    with gzip.open(vcf_file, 'rt') as fin, open('temporary_gt_file.txt', 'w') as fout:
+        for line in fin:
+            if line.startswith('##hailversion'):
+                # Write header line and update FORMAT column
+                fout.write(line)
+                # Add new FORMAT fields to the header
+                fout.write('##FORMAT=<ID=GT,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=ADFL,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=ADIR,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=ADSP,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=LC,Number=1,Type=Float,Description="">\n')
+                fout.write('##FORMAT=<ID=REPCI,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=REPCN,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=SO,Number=1,Type=String,Description="">\n')
+                fout.write('##FORMAT=<ID=QUAL,Number=1,Type=Float,Description="">\n')
+                fout.write('##INFO=<ID=END,Number=1,Type=Integer,Description="">\n')
+                fout.write('##INFO=<ID=REF,Number=1,Type=Integer,Description="">\n')
+                fout.write('##INFO=<ID=REPID,Number=1,Type=String,Description="">\n')
+                fout.write('##INFO=<ID=RL,Number=1,Type=Integer,Description="">\n')
+                fout.write('##INFO=<ID=RU,Number=1,Type=String,Description="">\n')
+                fout.write('##INFO=<ID=SVTYPE,Number=1,Type=String,Description="">\n')
+                fout.write('##INFO=<ID=VARID,Number=1,Type=String,Description="">\n')
+                fout.write('##ALT=<ID=STR1>\n')
+            elif line.startswith('##'):
+                # Write meta-information lines as they are
+                fout.write(line)
+            elif line.startswith('#CHROM'):
+                fout.write(line.replace('CPG', ''))
+
+            else:
+                # Process variant lines
+                parts = line.strip().split('\t')
+                parts[0] = 'chr' + parts[0]
+                info_field = parts[7]
+                format_field = parts[8]
+                sample_data = parts[9:]
+
+                # Extract PID from the sample data if available
+                pid_index = format_field.split(':').index('PID') if 'PID' in format_field.split(':') else None
+                pid = sample_data[0].split(':')[pid_index] if pid_index is not None else '.'
```

Comment:
Why would PID be unavailable?

Reply:
Yup; 

> Just clarifying that format field with have majority missing values except for GT?
> GT:ADFL:ADIR:ADSP:LC:REPCI:REPCN:SO:QUAL with an example of the sample data being 0/1:.:.:.:.:.:.:.:.
associaTR only pulls out the values from GT; so we can just leave the others blank rather than put in decoy values . 

>Why would PID be unavailable?
I don't know either, in the SNP VCF I see some with PIDs (like a unique string identifier that describes the locus), and others without.